web/app/scripts/custom-reports/custom-reports-modal-controller.js

```diff
@@ -0,0 +1,169 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function CustomReportsModalController($log, $modalInstance,
+                                          FilterState, GeographyState, QueryBuilder,
+                                          RecordState, RecordSchemaState) {
+        var ctl = this;
+        ctl.closeModal = closeModal;
+        ctl.createReport = createReport;
+
+        // Only the date part is needed when displaying these dates
+        ctl.dateFormat = 'MMM D, YYYY';
+
+        // These are set as the selections are made
+        ctl.colAggSelected = null;
+        ctl.rowAggSelected = null;
+        ctl.geoAggSelected = null;
+
+        // List of aggregations available for rows and columns.
+        // The time-related options defined here (type: `Time`) are constant.
+        // This list is later populated with all available filterable types (type: `Filter`),
+        // and all available geographies (type: `Geography`).
+        ctl.rowColAggs = [
+            {
+                label: 'Day of Month',
+                value: 'day',
+                type: 'Time'
+            },
+            {
+                label: 'Day of Week',
+                value: 'week_day',
+                type: 'Time'
+            },
+            {
+                label: 'Hour of Day',
+                value: 'hour',
+                type: 'Time'
+            },
+            {
+                label: 'Month of Year',
+                value: 'month',
+                type: 'Time'
+            },
+            {
+                label: 'Week of Year',
+                value: 'week',
+                type: 'Time'
+            }
```

Comment:
Oh, thanks, I missed that one. I think it's worth adding. If we see some performance issues while testing, we can always strip it out again.